Source/WebCore/html/HTMLDetailsElement.cpp

```diff
@@ -160,23 +160,53 @@ void HTMLDetailsElement::attributeChanged(const QualifiedName& name, const AtomS
                 root->appendChild(*m_defaultSlot);
                 queueDetailsToggleEventTask(DetailsState::Closed, DetailsState::Open);
                 if (auto& name = attributeWithoutSynchronization(nameAttr); document().settings().detailsNameAttributeEnabled() && !name.isEmpty()) {
-                    Vector<RefPtr<HTMLDetailsElement>> otherElementsInNameGroup;
-                    for (auto& detailsElement : descendantsOfType<HTMLDetailsElement>(rootNode())) {
-                        if (&detailsElement != this && detailsElement.attributeWithoutSynchronization(nameAttr) == name)
-                            otherElementsInNameGroup.append(&detailsElement);
-                    }
-                    for (RefPtr otherDetailsElement : otherElementsInNameGroup)
+                    bool shouldFireMutationEvents = document().shouldFireMutationEvents();
+                    document().setShouldFireMutationEvents(false);
+                    for (RefPtr otherDetailsElement : otherElementsInNameGroup())
                         otherDetailsElement->removeAttribute(openAttr);
+                    document().setShouldFireMutationEvents(shouldFireMutationEvents);
                 }
             } else {
                 root->removeChild(*m_defaultSlot);
                 queueDetailsToggleEventTask(DetailsState::Open, DetailsState::Closed);
             }
         }
-    } else
+    } else {
+        ensureDetailsExclusivityAfterMutation();
         HTMLElement::attributeChanged(name, oldValue, newValue, attributeModificationReason);
+    }
+}
+
+Node::InsertedIntoAncestorResult HTMLDetailsElement::insertedIntoAncestor(InsertionType insertionType, ContainerNode& parentOfInsertedTree)
+{
+    ensureDetailsExclusivityAfterMutation();
+    return HTMLElement::insertedIntoAncestor(insertionType, parentOfInsertedTree);
+}
+
+Vector<RefPtr<HTMLDetailsElement>> HTMLDetailsElement::otherElementsInNameGroup()
+{
+    Vector<RefPtr<HTMLDetailsElement>> otherElementsInNameGroup;
+    for (auto& detailsElement : descendantsOfType<HTMLDetailsElement>(rootNode())) {
```

Comment:
From a code readability standpoint, I think the single word local variable name `element` would be better than `detailsElement`.